fs/entry.go

```diff
@@ -79,50 +78,25 @@ type ErrorEntry interface {
 // ErrEntryNotFound is returned when an entry is not found.
 var ErrEntryNotFound = errors.New("entry not found")
 
-// ReaddirToIterate is an adapter for a naive Readdir -> IterateEntries implementation.
-func ReaddirToIterate(ctx context.Context, d Directory, cb func(context.Context, Entry) error) error {
-	entries, err := d.Readdir(ctx)
-	if err != nil {
-		return errors.Wrap(err, "error reading directory")
-	}
+// IterateEntriesAndFindChild iterates through entries from a directory and returns one by name.
+// This is a convenience function that may be helpful in implementations of Directory.Child().
+func IterateEntriesAndFindChild(ctx context.Context, d Directory, name string) (Entry, error) {
+	var result Entry
 
-	for _, e := range entries {
-		if err := cb(ctx, e); err != nil {
-			return err
+	if err := d.IterateEntries(ctx, func(c context.Context, e Entry) error {
+		if result == nil && e.Name() == name {
+			result = e
 		}
-	}
-
-	return nil
-}
-
-// IterateEntriesToReaddir is an adapter for a naive IterateEntries -> Readdir implementation.
-func IterateEntriesToReaddir(ctx context.Context, d Directory) (Entries, error) {
```

Comment:
it might make sense to keep this helper since there are already a few copies, but I would remove `Entries` and replace with `[]Entry` wherever needed. Perhaps call it `GetAllEntries(ctx, dir) ([]Entry, error)`

Reply:
idea: would it help if this helper returned `map[string]Entry`?

Reply:
yeah, let's get rid of `Entries` type and replace with `[]Entry` everywhere